### PR TITLE
Address fixmes in openapi spec

### PR DESCRIPTION
**Pull request by michel-laterman**
Address FIXME comments in openapi spec.

### CODE REVIEW

File: model/openapi.yml [hunk 1]
```diff
@@ -126,15 +127,17 @@ components:
         - metadata
       properties:
         type:
+          deprecated: true
```
Comment (michel-laterman) [inline model/openapi.yml:130]: Is there a plan to use different enrollment types or `shared_id`?
cc @cmacknz @blakerouse @michalpristas

Comment (cmacknz) [reply model/openapi.yml:130]: Not that I know about.

File: model/openapi.yml [hunk 2]
```diff
@@ -197,7 +205,8 @@ components:
           description: The ApiKey token that fleet-server has generated for the enrolling agent.
           type: string
           format: password
-        status: # FIXME Does this have any value? The UI does other calculations to infer status.
+        status:
+          deprecated: true
```
Comment (cmacknz) [inline model/openapi.yml:209]: I can't comment on the exact line, but while you are in here can you add `starting` as a state the agent can report to fleet?

https://github.com/elastic/fleet-server/blob/6e7ed9ebb20679c1463a1e8effd6c281d4f845d9/model/openapi.yml#L226-L233

**Review by joshdover — COMMENTED** [review]
Any reason not to just remove the deprecated fields completely?

File: model/openapi.yml [hunk 3]
```diff
@@ -162,30 +165,35 @@ components:
...
           #format: date-time
-        user_provided_metadata: # FIXME still needed?
+        user_provided_metadata:
+          deprecated: true
           description: A copy of the user provided metadata from the enrollment request. Currently will be empty.
           type: string
           format: application/json
           x-go-type: json.RawMessage
         local_metadata:
+          deprecated: true
```
[the remark this answers is not in the document]
Comment (michel-laterman) [reply model/openapi.yml:190]: oops, we do

File: model/openapi.yml (hunk 2 above)
Comment (michel-laterman) [reply model/openapi.yml:207]: added

File: model/openapi.yml (hunk 3 above)
[the remark this answers is not in the document]
Comment (michel-laterman) [reply model/openapi.yml:188]: This is the `local_metadata` attribute enrollResponse, not the enroll request; I don't think we need it here

[the remark this answers is not in the document]
Comment (joshdover) [reply model/openapi.yml:188]: Agreed

File: model/openapi.yml [hunk 4]
```diff
@@ -430,7 +443,8 @@ components:
         message:
           description: An acknowlegement message. The elastic-agent inserts the action ID and action type into this message.
           type: string
-        payload: # FIXME Deprecate and use data instead?
+        payload:
```
[the remark this answers is not in the document]
Comment (michel-laterman) [reply model/openapi.yml:459]: I'll hold off on any code changes in fleet-server & the agent for this as we are going to review how upgrades work in the near future.
the attribute will still be deprecated through

**Review by joshdover — APPROVED** [review]

**Review by cmacknz — APPROVED** [review]

**Pull request merged**